### DISCUSSION TITLE
Is there a way to map out the order of execution for migration files?

### DISCUSSION THREAD

**Discussion by joegiglio** [miguelgrinberg/Flask-Migrate#551] [General]
I am building a project with AI and it broke one of the migrations.  I am having a difficult time piecing it back together.

Is there a way to display the order of migrations from Initial Migration to the Latest?  The hash scheme does not go in any apparent order and I can not find this info without opening each file.  Thanks!

**Comment by miguelgrinberg** [comment]
Yes, you can look for the `down_revision` variable in each migration script. This points to the identifier of the previous migration. There is also a comment at the top that includes the creation date, which can also help.

**Comment by joegiglio** [comment]
I see that but it becomes a tangled mess if there any many migration files.  I found a command called `flask db history` and thought I hit the jackpot, but that is not in any order.  

I was hoping for output like:

Initial migration -> Next one -> Next one -> Last one

That does not exist?

**Reply by miguelgrinberg** [reply]
The history command shows the migrations in order, with the newest migration at the top, if I remember correctly. If you don't see the correct order there, then your migration history is corrupted.

**Comment by joegiglio** [comment]
Then it looks like it is corrupt. Any idea how that can happen from past experience?  

Once I piece it back together, how can I reset the schema?  I don't care about losing data.

**Reply by miguelgrinberg** [reply]
I don't understand what you are asking. If you don't care about preserving your migration history, then delete your migrations and create a single new migration, as if this was a new project.

**Comment by joegiglio** [comment]
Got it.  Thank you!
